pwa/src/components/entities/entityForm.tsx

```diff
@@ -236,7 +236,7 @@ export const EntityForm: React.FC<EntityFormProps> = ({ entityId }) => {
                             {entity !== null && entity.gateway !== undefined && entity.gateway !== null ? (
                               <SelectInputComponent
                                 options={sources}
-                                data={entity.gateway.name}
+                                data={entity.gateway.id}
```

Comment:
This doesn't work. The old implementation didn't work either. 

The reason behind this is that the `SelectInputComponent` is a bit complex to grasp, but the fix is as followed:

1. The `options` prop we're sending needs a valid `name` and `value` for the component to work and then
2. We need to set the current value in the `value` prop (which has to be the same as `option.value` here to be selected by default..

So here's how we do that:

1. We need to `.map()` the sources that we `GET` from the backend, like so:

```
const _sources = res.data.map((source) => ({
  name: source.name,
  value: `/admin/attributes/${source.id}`,
  }));
setSources(_sources);

//NOTE: this was just setSources(res.data); but now we're mapping the data to the structure we need in the selectelement!
```

2. We need to update the `SelectInputComponent` implementation, like so:

```
<SelectInputComponent
                            options={sources}
                            data={`/admin/attributes/${entity?.gateway?.id}`}
                            name={"gateway"}
                            id={"gatewayInput"}
                            nameOverride={"Source"}
                          />
```